tst/com/amazon/ion/benchmark/dataGeneratorTest.java

```diff
@@ -0,0 +1,202 @@
+        }
+    }
+
+    /**
+     * Assert the exponent range of generated Ion decimals is conform with the expected range.
+     * @throws Exception if error occurs when executing Ion data generator.
+     */
+    @Test
+    public void testGeneratedDecimalExponentRange() throws Exception {
+        Map<String, Object> optionsMap = dataGeneratorTest.testCasesGenerator("1000","decimal", "[1,1114111]", "[1,10]", "test2.10n", "2021T", "[ion_binary]", "[-13,-2]");
+
+        GeneratorOptions.executeGenerator(optionsMap);
+        IonReader reader = IonReaderBuilder.standard().build(new BufferedInputStream(new FileInputStream(optionsMap.get("<output_file>").toString())));
+
+        List<Integer> range = WriteRandomIonValues.rangeParser(optionsMap.get("--decimal-exponent-range").toString());
+        while (reader.next() != null) {
+            int exp = reader.decimalValue().scale();
+            assertTrue(exp * (-1) >= range.get(0) && exp * (-1) <= range.get(1));
+        }
+    }
+
+    /**
+     * Assert the range of coefficient digits number of generated Ion decimals is conform with the expected range.
+     * @throws Exception if error occurs when executing Ion data generator.
+     */
+    @Test
+    public void testGeneratedDecimalCoefficientRange() throws Exception {
+        Map<String, Object> optionsMap = dataGeneratorTest.testCasesGenerator("3000","decimal", "[1,1114111]", "[1,4]", "tes3.10n", "2021T", "[ion_text]", "[-13,-2]");
+        GeneratorOptions.executeGenerator(optionsMap);
+        IonReader reader = IonReaderBuilder.standard().build(new BufferedInputStream(new FileInputStream(optionsMap.get("<output_file>").toString())));
+        List<Integer> range = WriteRandomIonValues.rangeParser(optionsMap.get("--decimal-coefficient-digit-range").toString());
+
+        while (reader.next() != null) {
+            BigInteger coefficient = reader.decimalValue().unscaledValue();
+            double factor = Math.log(2) / Math.log(10);
+            int digitCount = (int) (factor * coefficient.bitLength() + 1);
+            if (BigInteger.TEN.pow(digitCount - 1).compareTo(coefficient) > 0) {
+                digitCount = digitCount - 1;
+            }
+            assertTrue(digitCount >= range.get(0) && digitCount <= range.get(1));
+        }
+    }
+
+    /**
+     * Assert the format of generated file is conform with the expected format [ion_binary|ion_text].
+     * @throws Exception if error occurs when executing Ion data generator.
+     */
+    @Test
+    public void testGeneratedFormat() throws Exception {
+        Map<String, Object> optionsMap = dataGeneratorTest.testCasesGenerator("500","symbol", "[1,1114111]", "[1,4]", "test11.ion", "2021T", "[ion_text]", "[-13,-2]");
+
+        GeneratorOptions.executeGenerator(optionsMap);
+        String format = optionsMap.get("--format").toString().substring(1, optionsMap.get("--format").toString().length() - 1);
```

Comment:
`--format` should just be specified as `ion_text`, avoiding the need for this trimming.